dask/layers.py

```diff
@@ -0,0 +1,837 @@
+import tlz as toolz
+import operator
+from collections import defaultdict
+from importlib import import_module
+
+from .core import keys_in_tasks
+from .utils import apply, insert, stringify, stringify_collection_keys
+from .highlevelgraph import Layer
+
+# TODO: Make sure dask.blockwise does not require any
+# imports beyond minimum dependencies
+from .blockwise import Blockwise, BlockwiseIODeps, blockwise_token
+
+
+#
+##
+###  General Utilities
+##
+#
+
+
+class CallableLazyImport:
+    """Function Wrapper for Lazy Importing"""
+
+    def __init__(self, function_path):
+        split_path = function_path.split(".")
+        self.module = ".".join(split_path[:-1])
+        self.name = split_path[-1]
+
+    def __call__(self, *args, **kwargs):
+        func = getattr(import_module(self.module), self.name)
+        return func(*args, **kwargs)
```

Comment:
This looks new. Just to confirm, this is so we can materialize the graph on the scheduler without importing, say, `pandas`?

Reply:
Right - I am very interested in feedback on this. The idea is to have a "callable" at graph materialization time without requiring the scheduler to acutally import anything from a something like pandas/numpy/dask.dataframe/dask.array.

Reply:
Hmm - I actually think we can do that with a small tweak (by adding a deserializing parameter, as we do in Blockwise). I’ll try this out tonight.